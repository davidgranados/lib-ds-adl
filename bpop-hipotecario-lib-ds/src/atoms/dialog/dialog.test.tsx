import React from "react";
import { shallow } from "enzyme";

import Dialog from "./dialog";

describe("Dialog", () => {
  let show = true;
  test("Dialog should renders ok", () => {
    const component = shallow(
      <Dialog
        show={show}
        onToggle={(s) => {
          show = s;
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test("Dialog should hide", () => {
    show = true;
    const component = shallow(
      <Dialog
        show={show}
        onToggle={(s) => {
          show = s;
        }}
      />
    );
    component.find("#dialog-wrapper-overflow").simulate("click");
    expect(show).toBe(false);
  });

  test("Dialog should show", () => {
    show = true;
    shallow(
      <Dialog
        show={show}
        onToggle={(s) => {
          show = s;
        }}
      />
    );
    expect(show).toBe(true);
  });

  test("Dialog should show", () => {
    shallow(
      <Dialog
        show={show}
        dataTestId={"testid"}
        onToggle={(s) => {
          show = s;
        }}
      />
    );
    expect(show).toBe(true);
  });
});
