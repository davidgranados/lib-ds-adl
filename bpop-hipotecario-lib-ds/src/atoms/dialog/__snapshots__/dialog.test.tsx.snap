// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Dialog should renders ok 1`] = `
<div
  className="wrapper show false"
  data-testid="dialog-wrapper"
  id="dialog-wrapper"
>
  <div
    className="overflow"
    data-testid="dialog-wrapper-overflow"
    id="dialog-wrapper-overflow"
    onClick={[Function]}
  />
  <div
    className="modal"
    style={
      Object {
        "width": undefined,
      }
    }
  />
</div>
`;
